pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/user/UserDao.java

```diff
@@ -110,6 +112,26 @@ public int create(UserDto userDto) {
         });
     }
 
+    public static void update(int userId, UserDto userDto) {
+        String email = userDto.getEmailOrThrow();
+        String errorMsg = "Error updating user " + email;
+        int res = inTransaction(conn -> {
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_UPDATE_USER)) {
+                stmt.setString(1, userDto.getName().orElse(null));
```

Comment:
Since its updating existing user, aren't we nullifying the existing name if NOT passed ?
looks like in UserAdminService we are checking for name to be non null though

Reply:
Yup, I chose to implement it this way (the caller determines what to update by looking at the existing record) rather than doing a selective update based on the DTO. But I can change that if there is already a convention in place. 

I think you are suggesting that the logic in the caller only include the fields to update, and the logic here would be to only update the non-null fields. One downside is that you cannot set a field to null with the update.

Reply:
In that case we can just dostmt.setString(1, userDto.getName());
If caller sets/passes null, update as null.  End result will be same as existing code I guess.

Reply:
I think we should decide on the behavior, then consider the implementation.
In this case it is either: assign whatever the DTO has, including NULLs, which is the implementation in this PR; or only assign those values that are not null. The latter seems problematic since it does not provide the caller a way to assign NULL to a field.

What do you think?

This is an unusual DTO since its get methods return optional, which is not our convention. If that seems confusing we can change that, but perhaps we can do that as a separate PR?

Reply:
assign whatever the DTO has, including NULLs seems good to me.